packages/syft/src/syft/core/adp/data_subject_ledger.py

```diff
@@ -375,29 +401,31 @@ def _get_batch_rdp_constants(
         return query_constants
 
     def _get_epsilon_spend(self, rdp_constants: np.ndarray) -> np.ndarray:
-        # rdp_constants_lookup = (rdp_constants - 1).astype(np.int64)
         rdp_constants_lookup = convert_constants_to_indices(rdp_constants)
-        try:
-            # needed as np.int64 to use take
-            eps_spend = jax.jit(jnp.take)(
-                self._cache_constant2epsilon, rdp_constants_lookup
-            )
+        if rdp_constants_lookup.max() - len(self._cache_constant2epsilon) >= 150_000:
+            eps_spend = self._fetch_eps_spend_for_big_rdp(rdp_constants)
```

Comment:
MAYBE: We could pass rdp_constants lookup, to the `_fetch_eps_spend_for_big_rdp` function to save on recalculating the `cache_indices` variable.